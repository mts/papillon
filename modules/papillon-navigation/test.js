import React from 'react';
import {
  menu,
  menuWithIconAvatarCounter,
  menuWithHeading
} from './flavorMenu';
import {
  underlineNav,
  underlineNavActions
} from './flavorUnderlineNav';

describe('<Navigation />', () => {
  let wrapper;

  describe('Instance', () => {
    test('must be an instance of Menu', () => {
      wrapper = renderMount(menu);
      expect(wrapper.find('Menu')).toHaveLength(1);
    });

    test('must be an instance of UnderlineNav', () => {
      wrapper = renderMount(underlineNav);
      expect(wrapper.find('UnderlineNav')).toHaveLength(1);
    });
  });

  describe('Snaphot', () => {
    test('must match menu', () => {
      expect(global.renderToJSON(menu)).toMatchSnapshot();
    });

    test('must match menuWithIconAvatarCounter', () => {
      expect(global.renderToJSON(menuWithIconAvatarCounter)).toMatchSnapshot();
    });

    test('must match menuWithHeading', () => {
      expect(global.renderToJSON(menuWithHeading)).toMatchSnapshot();
    });

    test('must match underlineNav', () => {
      expect(global.renderToJSON(underlineNav)).toMatchSnapshot();
    });

    test('must match underlineNavActions', () => {
      expect(global.renderToJSON(underlineNavActions)).toMatchSnapshot();
    });
  });
});
