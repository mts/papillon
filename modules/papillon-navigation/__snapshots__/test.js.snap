// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Snaphot must match menu 1`] = `
<nav
  aria-label="Person settings"
  className="menu"
>
  <a
    aria-current="page"
    className="menu-item selected"
    href="#url"
  >
    <span
      className=""
    >
      Account
    </span>
  </a>
  <a
    className="menu-item"
    href="#url"
  >
    <span
      className=""
    >
      Profile
    </span>
  </a>
  <a
    className="menu-item"
    href="#url"
  >
    <span
      className=""
    >
      Emails
    </span>
  </a>
  <a
    className="menu-item"
    href="#url"
  >
    <span
      className=""
    >
      Notifications
    </span>
  </a>
</nav>
`;

exports[`<Navigation /> Snaphot must match menuWithIconAvatarCounter 1`] = `
<nav
  aria-label="Person settings"
  className="menu"
>
  <a
    aria-current="page"
    className="menu-item selected"
    href="#url"
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 16 16"
      width={16}
    >
      <path
        d="M4.48 7.27c.26.26 1.28 1.33 1.28 1.33l.56-.58-.88-.91 1.69-1.8s-.76-.74-.43-.45c.32-1.19.03-2.51-.87-3.44C4.93.5 3.66.2 2.52.51l1.93 2-.51 1.96-1.89.52-1.93-2C-.19 4.17.1 5.48 1 6.4c.94.98 2.29 1.26 3.48.87zm6.44 1.94l-2.33 2.3 3.84 3.98c.31.33.73.49 1.14.49.41 0 .82-.16 1.14-.49.63-.65.63-1.7 0-2.35l-3.79-3.93zM16 2.53L13.55 0 6.33 7.46l.88.91-4.31 4.46-.99.53-1.39 2.27.35.37 2.2-1.44.51-1.02L7.9 9.08l.88.91L16 2.53z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="ml-1"
    >
      Account
    </span>
  </a>
  <a
    className="menu-item"
    href="#url"
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 12 16"
      width={12}
    >
      <path
        d="M12 14.002a.998.998 0 0 1-.998.998H1.001A1 1 0 0 1 0 13.999V13c0-2.633 4-4 4-4s.229-.409 0-1c-.841-.62-.944-1.59-1-4 .173-2.413 1.867-3 3-3s2.827.586 3 3c-.056 2.41-.159 3.38-1 4-.229.59 0 1 0 1s4 1.367 4 4v1.002z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="ml-1"
    >
      Profile
    </span>
  </a>
  <a
    className="menu-item"
    href="#url"
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 14 16"
      width={14}
    >
      <path
        d="M0 4v8c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1H1c-.55 0-1 .45-1 1zm13 0L7 9 1 4h12zM1 5.5l4 3-4 3v-6zM2 12l3.5-3L7 10.5 8.5 9l3.5 3H2zm11-.5l-4-3 4-3v6z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="ml-1"
    >
      Emails
    </span>
  </a>
  <a
    className="menu-item"
    href="#url"
  >
    <svg
      aria-hidden="true"
      className="octicon"
      height={16}
      role="img"
      style={
        Object {
          "display": "inline-block",
          "fill": "currentColor",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 16 16"
      width={16}
    >
      <path
        d="M4.78 5.78c.25-.25.25-.67 0-.92-.32-.33-.48-.76-.48-1.19 0-.43.16-.86.48-1.19.25-.26.25-.67 0-.92a.613.613 0 0 0-.45-.19c-.16 0-.33.06-.45.19-.57.58-.85 1.35-.85 2.11 0 .76.29 1.53.85 2.11.25.25.66.25.9 0zM2.32.19a.651.651 0 0 0-.92 0C.47 1.15 0 2.41 0 3.66c0 1.26.47 2.52 1.4 3.48.25.26.66.26.91 0s.25-.68 0-.94c-.68-.7-1.02-1.62-1.02-2.54 0-.92.34-1.84 1.02-2.54a.66.66 0 0 0 .01-.93zm5.69 5.1a1.62 1.62 0 1 0-1.62-1.62c-.01.89.72 1.62 1.62 1.62zM14.58.2a.628.628 0 0 0-.91 0c-.25.26-.25.68 0 .94.68.7 1.02 1.62 1.02 2.54 0 .92-.34 1.83-1.02 2.54-.25.26-.25.68 0 .94a.651.651 0 0 0 .92 0c.93-.96 1.4-2.22 1.4-3.48A5.048 5.048 0 0 0 14.58.2zM8.01 6.59c-.41 0-.83-.1-1.2-.3l-3.15 8.37h1.49l.86-1h4l.84 1h1.49L9.2 6.29c-.38.2-.78.3-1.19.3zM8 7.07l1.01 3.6h-2L8 7.07zm-1.99 5.59l1-1h2l1 1h-4zm5.19-11.1c-.25.25-.25.67 0 .92.32.33.48.76.48 1.19 0 .43-.16.86-.48 1.19-.25.26-.25.67 0 .92a.63.63 0 0 0 .9 0c.57-.58.85-1.35.85-2.11 0-.76-.28-1.53-.85-2.11a.634.634 0 0 0-.9 0z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="ml-1"
    >
      <span
        className="Counter"
      >
        3
      </span>
      Notifications
    </span>
  </a>
</nav>
`;
